docs/features/async-main.md

```diff
@@ -0,0 +1,12 @@
+# Async Task Main
+## [dotnet/csharplang proposal](https://github.com/dotnet/csharplang/blob/master/proposals/async-main.md)
+
+## Technical Details
+
+* The compiler must recognize `Task` and `Task<int>` as valid entrypoint return types in addition to `void` and `int`.
```

Comment:
Btw, @TyOverby, you may want to record links to relevant LDM notes to your feature doc, once they get posted.